spec/lib/vanagon/engine/pooler_spec.rb

```diff
@@ -21,28 +21,86 @@
   before :each do
     # suppress `#warn` output during tests
     allow_any_instance_of(Vanagon::Platform::DSL).to receive(:warn)
+
+    # stubbing ENV doesn't actually intercept ENV because ENV
+    # is special and you aren't. So we have to mangle the value
+    # of ENV around each of these tests if we want to prevent
+    # actual user credentials from being loaded, or attempt to
+    # validate the way that credentials are read.
+    ENV['REAL_HOME'] = ENV.delete 'HOME'
+    ENV['HOME'] = Dir.mktmpdir
+
+    # This should help maintain the ENV['HOME'] masquerade
+    allow(Dir).to receive(:home).and_return(ENV['HOME'])
+  end
+
+  after :each do
+    # Altering ENV directly is a legitimate code-smell.
+    # We should at least clean up after ourselves.
+    ENV['HOME'] = ENV.delete 'REAL_HOME'
   end
 
+  # We don't want to run the risk of reading legitimate user
+  # data from a user who is unfortunate enough to be running spec
+  # tests on their local machine. This means we have to intercept
+  # and stub out a very specific environment, where:
+  #   1) an environment variable exists
+  #   2) the env. var is unset, and only a ~/.vmpooler-token file exists
+  #   3) the env. var is unset, and only a ~/.vmfloaty.yml file exists
   describe "#load_token" do
-    after(:each) { ENV['VMPOOLER_TOKEN'] = nil }
+    let(:environment_value) { 'cafebeef' }
+    let(:token_value) { 'decade' }
+    let(:pooler_token_file) { File.expand_path('~/.vanagon-token') }
+    let(:floaty_config) { File.expand_path('~/.vmfloaty.yml') }
+
+    it 'prefers the VMPOOLER_TOKEN environment variable to a config file' do
+      stub_const('ENV', { 'VMPOOLER_TOKEN' => environment_value } )
```

Comment:
Yeah, maybe. Since this test *only* focuses on the value of that env. var. though, does it matter? I'm open to suggestions that don't involve setting and unsetting any more env. values directly.